src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -1211,24 +1211,21 @@ LedgerMaster::findNewLedgersToPublish(std::unique_lock<std::recursive_mutex>& sl
                 << "Trying to fetch/publish valid ledger " << seq;
 
             std::shared_ptr<Ledger const> ledger;
-            // This can throw
+            // This can throw.
             auto hash = hashOfSeq(*valLedger, seq, m_journal);
-            // VFALCO TODO Restructure this code so that zero is not
-            // used.
-            if (! hash)
-                hash = beast::zero; // kludge
-            if (seq == valSeq)
-            {
-                // We need to publish the ledger we just fully validated
-                ledger = valLedger;
-            }
-            else if (hash->isZero())
+            // VFALCO TODO Restructure this code so that zero is not used.
+            if (!hash || hash->isZero())
```

Comment:
I think that we can probably just fix this TODO for Vinnie.

Reply:
Are you suggesting to just remove it? I don't know enough to confirm that `hashOfSeq` never returns zero.

Reply:
I _think_ we've already fixed that but on second thought, I'm not 100% sure, so let's not do that right now.